### DISCUSSION TITLE
Adding Jalali months instead of Gregorian months in jalali calendar

### DISCUSSION THREAD

**Discussion by kianlatifi** [gpbl/react-day-picker#2571] [Translation and localization]
in the example at https://daypicker.dev/docs/localization#jalali-calendar , the months are based on Gregorian calendar not Jalali calendar.

**Comment by gpbl** [comment]
Hello @kianlatifi - I’m not familiar with the Jalali calendar. Could you please guide me on how to convert it from the Gregorian calendar?

**Reply by kianlatifi** [reply]
of course!
you have experimental support for it and I saw that the months aren't Jalali.
list of Jalali months:
Spring
1. Farvardin (31d)
2. Ordibehesht (31d)
3. Khordad (31d)
Summer
4. Tir (31d)
5. Mordad (31d)
6. Shahrivar (31d)
Autumn
7. Mehr (30d)
8. Aban (30d)
9. Azar (30d)
Winter
10. Dey (30d)
11. Bahman (30d)
12. Esfand (29d or 30d)

Every four year it's leap year.

to convert Gregorian to Jalali and reverse it you can read these stackoverflow issues:
https://stackoverflow.com/questions/35570884/how-to-change-gregorian-date-to-persian-date-in-javascript
https://stackoverflow.com/questions/54838700/how-to-convert-persian-date-to-gregorian-date

**Reply by gpbl** [reply]
@kianlatifi thanks for your help!

- Could you please point out what’s working and what’s not working in the current `react-day-picker/jalali`?
- DayPicker offers `formatters`, `labels`, and a [`dateLib` prop](https://daypicker.dev/api/interfaces/PropsBase#datelib) to customize how dates are displayed in the calendar. How would you use these features to render a Jalali calendar?
- We rely on [date-fns-jalali](https://www.npmjs.com/package/date-fns-jalali) functions to render the calendar. Is this sufficient? Are there any bugs in our implementation? I’d appreciate your insights here.
- I understand from Wikipedia that there’s a distinction between [Solar Hijri](https://en.wikipedia.org/wiki/Solar_Hijri_calendar) and Jalali calendar, with the former being the “easier” version adopted in Persia. I’m now wondering if `date-fns-jalali` follows Hijri or Jalali 🤔

**Reply by kianlatifi** [reply]
maybe I'm wrong but Solar Hijri and Jalali are the same, I haven't tested it yet and by just looking at the doc, I found out the month labels are not based on Jalali (I mentiond the months names but if you want them in Persian notation here you go:
    فروردین 
    اردیبهشت 
    خرداد 
    تیر 
    مرداد 
    شهریور 
    مهر 
    آبان 
    آذر 
    دی 
    بهمن 
    اسفند 
)
and I think the date-fns-jalali package is decent 
unfortunately I'm very busy these days but if I can, I will review the Jalali section more thoroughly. in the mean time, feel free to ask me any other things about the Jalali section.

**Reply by gpbl** [reply]
I understood that the issue isn’t just the names of the months. Also, the years are numbered differently, and from what I read, the numbering of days and months is different as well:

> Hijri Calendar: It has 354 or 355 days, with 12 months of 29 or 30 days each. This is shorter than a solar year.

https://github.com/gpbl/react-day-picker/pull/2574 is an attempt to rewrite the date manipulation library using `hijri-date`. It didn’t work initially, but I’d like someone to take a look at it.
